panther_driver/src/panther_hardware.py

```diff
@@ -0,0 +1,212 @@
+        self.timer_charger = rospy.Timer(rospy.Duration(0.5), self.publish_charger_state)
+
+        self.first_start_sequence()
+
+        rospy.spin()
+
+    def first_start_sequence(self):
+        """
+        First start sequence for motors which is meant to power up modules in correct order
+        """
+        GPIO.output(VMOT_ON, 1)
+        time.sleep(0.5)
+
+        GPIO.output(DRIVER_EN, 1)
+        time.sleep(0.2)
+
+        GPIO.output(AUX_PW_EN, 1)
+
+
+    def publish_e_stop_state(self, event=None):
+        msg = Bool()
+        msg.data = self.read_e_stop_pin()
+        self.e_stop_state_pub.publish(msg)
+
+    def publish_charger_state(self, event=None):
+        msg = Bool()
+        msg.data = GPIO.input(CHRG_SENSE)
+        self.charger_state_pub.publish(msg)
+
+    def handle_aux_power_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, AUX_PW_EN, "Aux power enable")
+
+    def handle_charger_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, CHRG_EN, "Charger enable")
+
+    def handle_disable_digital_power(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, VDIG_OFF, "Digital power disable")
+
+    def handle_motors_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, DRIVER_EN, "Motors driver enable")
+
+    def handle_fan_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, FAN_SW, "Fan enable")
+
+    
+    def handle_trigger_e_stop(self, req: TriggerRequest):
+        self.watchdog.turn_off()
+        return TriggerResponse(True, f"E-STOP triggered, watchdog turned off")
+
+    def handle_reset_e_stop(self, req: TriggerRequest):
+        # Read value before reset
+        e_stop_initial_val = self.read_e_stop_pin()
+
+        # Check if E-STOP reset is needed
+        print(e_stop_initial_val)
+        if e_stop_initial_val == False:
+            msg = "E-STOP was not triggered, reset is not needed"
+            response = TriggerResponse(False, msg)
+            return response
+
+        # Switch e-stop pin to output
+        GPIO.setup(E_STOP_RESET, GPIO.OUT)
+
+        #Perform reset
+        success = self.handle_gpio_write(True, E_STOP_RESET, "E-STOP reset")
+        self.watchdog.turn_on()
+        time.sleep(0.1)
+
+        # Switch back to input after writing
+        GPIO.setup(E_STOP_RESET, GPIO.IN)
+
+        # Check if E-STOP reset succeeded
+        e_stop_val = self.read_e_stop_pin()
+        print(f"E-STOP state = {e_stop_val}")
+
+        # Send correct response
+        if e_stop_initial_val == e_stop_val or not success:
+            msg = "E-STOP reset not successful, initial value is the same as after reset"
+            response = TriggerResponse(False, msg)
+            return response
+
+        msg = "E-STOP reset successful"
+        response = TriggerResponse(success, msg)
+        return response
```

Comment:
I wait with reviewing this method till you decide on the message type here

Reply:
The message type will stay. We discussed it with other tam members. This service is not as simple as toggling one pin